### ISSUE TITLE
Better first-class support for ash

### ISSUE THREAD

**Issue by mcandre** [koalaman/shellcheck#1841]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env ash
echo "Hello World!"
```

#### Here's what shellcheck currently says:

```console
$ shellcheck hello

In hello line 1:
#!/usr/bin/env ash
^-- SC2187: Ash scripts will be checked as Dash. Add '# shellcheck shell=dash' to silence.
```

#### Here's what I wanted or expected to see:

I expect to see no output. But ShellCheck currently only handles dash scripts, not pure ash scripts.

While ShellCheck does offer a configurable SC2187 rule as a stop-gap, I think in the long run it would be more helpful to enhance ash support so this warning can be dropped. Meanwhile, I am inserting `# shellcheck shell=ash` in my scripts.
